cni/network/network.go

```diff
@@ -855,3 +890,120 @@ func (plugin *netPlugin) Update(args *cniSkel.CmdArgs) error {
 
 	return nil
 }
+
+// Temporary function to determine whether we need to disable SNAT due to NMAgent support
+func determineSnat(nwCfg *cni.NetworkConfig) (bool, error) {
+	var (
+		snatConfig            snatConfiguration
+		retrieveSnatConfigErr error
+		jsonFile              *os.File
+		httpClient            = &http.Client{Timeout: time.Second * 5}
+		snatConfigFile        = snatConfigFileName + jsonFileExtension
+	)
+
+	// Check if we've already retrieved NMAgent version and determined whether to disable snat on host
+	if jsonFile, retrieveSnatConfigErr = os.Open(snatConfigFile); retrieveSnatConfigErr == nil {
+		bytes, _ := ioutil.ReadAll(jsonFile)
+		jsonFile.Close()
+		if retrieveSnatConfigErr = json.Unmarshal(bytes, &snatConfig); retrieveSnatConfigErr != nil {
+			log.Errorf("[cni-net] failed to unmarshal to snatConfig with error %v",
+				retrieveSnatConfigErr)
+		}
+
+	}
+
+	// If we weren't able to retrieve snatConfiguration, query NMAgent
+	if retrieveSnatConfigErr != nil {
+		var resp *http.Response
+		resp, retrieveSnatConfigErr = httpClient.Get(nmAgentSupportedApisURL)
+		if retrieveSnatConfigErr == nil {
+			defer resp.Body.Close()
+
+			if resp.StatusCode == http.StatusOK {
+				var bodyBytes []byte
+				// if the list of APIs (strings) contains the nmAgentSnatSupportAPI we will disable snat on host
+				if bodyBytes, retrieveSnatConfigErr = ioutil.ReadAll(resp.Body); retrieveSnatConfigErr == nil {
+					bodyStr := string(bodyBytes)
+					if !strings.Contains(bodyStr, nmAgentSnatAndDnsSupportAPI) {
+						snatConfig.EnableSnatForDns = true
+						snatConfig.EnableSnatOnHost = !strings.Contains(bodyStr, nmAgentSnatSupportAPI)
+					}
+
+					if retrieveSnatConfigErr = acquireSnatConfigLock(); retrieveSnatConfigErr == nil {
+						jsonStr, _ := json.Marshal(snatConfig)
+						fp, err := os.OpenFile(snatConfigFile, os.O_CREATE|os.O_WRONLY|os.O_TRUNC, os.FileMode(0664))
+						if err == nil {
+							fp.Write(jsonStr)
+							fp.Close()
+						} else {
+							log.Printf("[cni-net] failed to save snatConfig")
+						}
+						releaseSnatConfigLock()
+					}
+				}
+			} else {
+				retrieveSnatConfigErr = fmt.Errorf("nmagent request status code %d", resp.StatusCode)
+			}
+		}
+	}
+
+	// Log and return the error when we fail acquire snat configuration for host and dns
+	if retrieveSnatConfigErr != nil {
+		log.Errorf("[cni-net] failed to acquire SNAT configuration with error %v",
+			retrieveSnatConfigErr)
+		return snatConfig.EnableSnatForDns, retrieveSnatConfigErr
+	}
+
+	nwCfg.EnableSnatOnHost = snatConfig.EnableSnatOnHost
```

Comment:
can we return both snatConfig.EnableSnatForDns and snatConfig.EnableSnatOnHost  from this function and set nwCfg outside this function..so that it remains consistent

Reply:
EnableSnatForDns is not a part of nwCfg though